dataflow/src/main/java/org/checkerframework/dataflow/cfg/CFGBuilder.java

```diff
@@ -4878,8 +4878,34 @@ public Node visitWhileLoop(WhileLoopTree tree, Void p) {
 
             // Condition
             addLabelForNextNode(conditionStart);
+
+            boolean isWhileTrue = false;
+            int firstConditionPos = nodeList.size();
```

Comment:
Instead of going through the converted nodes, wouldn't it be easier to just look at the condition tree and see whether it's the literal `true`?